File: CPP/hybridinh.cpp
#include<iostream>
using namespace std;
class student
{
private:
    int id;
    string name;
public:
    void setdata()
    {
        cout<<"enter id and name"<<endl;
        cin>>id>>name;
    }
    void showdata()
    {
        cout<<id<<"\t"<<name;
    }
};
class mark:public student
{
private:
    float m1,m2;
public:
    void setmarks()
    {
        cout<<"enter two subject marks";
        cin>>m1>>m2;
    }
    float getm1()
    {
        return m1;

    }
    float getm2()
    {
        return m2;
    }
};
class sport
{
private:
    int grade;
public:
    void setgrade()
    {
        cout<<"\n enter sport grade"<<endl;
        cin>>grade;
    }
    void showgrade()
    {
        cout<<grade<<endl;
    }
};
class result:public mark,public sport
{
public:
        void set()
        {
            setdata();
            setmarks();
            setgrade();
        }
        void showresult()
        {
            showdata();
            float avg;
            avg=(getm1()+getm2())/2;
            cout<<"average mark are:"<<avg<<endl;
            showgrade();
        }
};
int main()
{
    result ob;
    ob.set();
    ob.showresult();
    return(0);
}

File: PYTHON/MyProject1/hybridinh.py
class student:
    def setdata(self):
        self.name=input("enter name:")
        self.id=float(input("enter id:"))

    def showdata(self):
        print(self.id,self.name)

class mark(student):
    def setmarks(self):
        self.m1 = float(input("enter mark1:"))
        self.m2 = float(input("enter mark2:"))
    def showmarks(self):
        print(self.m1,self.m2)

class sport:
    def setgrade(self):
        self.grade=int(input("enter grade:"))
    def showgrade(self):
        print(self.grade)

class result(mark,sport):
    def set(self):
        self.setdata()
        self.setmarks()
        self.setgrade()
    def diplayresult(self):
        self.showdata()
        avg=(self.m1+self.m2)/2
        self.showmarks()
        print("average of marks:",avg)
        self.showgrade()


ob1=result()
ob1.set()
ob1.diplayresult()
